clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/model/AmazonReservationReport.groovy

```diff
@@ -61,22 +61,49 @@ class AmazonReservationReport implements ReservationReport {
     OperatingSystemType os
     AtomicInteger totalReserved = new AtomicInteger(0)
     AtomicInteger totalUsed = new AtomicInteger(0)
+    AtomicInteger totalBorrowed = new AtomicInteger(0)
 
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
     Map<String, AccountReservationDetail> accounts = new ConcurrentHashMap<>()
 
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    List<Borrow> borrows = []
+
+    @JsonProperty
+    String id() {
+      return "${region}:${availabilityZoneId()}:${instanceType}:${os.name.toLowerCase()}"
+    }
+
+    @JsonProperty
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    Integer totalBorrowed() {
+      // returning null will cause this attribute to not be serialized
+      // (will only appear in v3 report for instance types that have related borrows
+      return totalBorrowed.intValue() ?: null
```

Comment:
This might be better off as a jackson `View` that excludes `totalBorrowed` from the v1/v2 reports.

I'll investigate that.